pkg/k8shandler/statefulset.go

```diff
@@ -242,15 +248,15 @@ func (node *statefulSetNode) restart(upgradeStatus *api.ElasticsearchNodeStatus)
 
 	if upgradeStatus.UpgradeStatus.UpgradePhase == api.NodeRestarting {
 
-		err, ordinal := node.partition()
+		ordinal, err := node.partition()
 		if err != nil {
 			logrus.Infof("Unable to get node ordinal value: %v", err)
 			return
 		}
 
 		for index := ordinal; index > 0; index-- {
 			// get podName based on ordinal index and node.name()
-			podName := fmt.Sprintf("%v-%v", node.name(), index)
+			podName := fmt.Sprintf("%v-%v", node.name(), index-1)
```

Comment:
since this is off by `1` given we set the partition to `replicas` at first, can we just do `index := ordinal - 1` instead of `index-1`?